02-OOP-Inheritance/Ex05/src/classMovie.js

```diff
@@ -0,0 +1,26 @@
+import {eventEmitter} from './classEventEmitter';
+
+class Movie extends eventEmitter {
+	constructor(title, year, duration) {
+		super();
+		this.title = title;
+		this.year = year;
+		this.duration = duration;
+		this.cast = [];
+	}
+	play() {
+		super.emit('play');
+		this.played = true;
```

Comment:
You don't need to add a flag to check for a movie played. When you set Logger.log as callback of the event play, execute the event, will produce the Logger.log to be executed too.